File: hrms-frontend/src/services/holidayService .js
import axiosClient from "./axiosClient";

const holidayService = {
  getAllHolidays: () => {
    return axiosClient.get("/holidays");
  },

  createHoliday: (holidayPayload) => {
    return axiosClient.post("/holidays", holidayPayload);
  },

  checkIfHoliday: (date) => {
    return axiosClient.get(`/holidays/check/${date}`);
  },
  getHolidayById: (id) => {
    return axiosClient.get(`/holidays/${id}`);
  },

  updateHoliday: (id, holidayPayload) => {
    return axiosClient.put(`/holidays/${id}`, holidayPayload);
  },
  deleteHoliday: (id) => {
    return axiosClient.delete(`/holidays/${id}`);
  },
};

export default holidayService;

File: hrms-backend/src/main/java/sep490/com/example/hrms_backend/service/HolidayService.java
package sep490.com.example.hrms_backend.service;

import sep490.com.example.hrms_backend.dto.HolidayDTO;

import java.time.LocalDate;
import java.util.List;

public interface HolidayService {
    List<HolidayDTO> getAllHolidays();
    HolidayDTO createHoliday(HolidayDTO holidayDTO);
    boolean isHoliday(LocalDate date);

    void softDeleteHoliday(Long id);

    HolidayDTO getHolidayById(Long id);
    HolidayDTO updateHoliday(Long id, HolidayDTO holidayDTO);
}
